src/prometheus_middleware.rs

```diff
@@ -61,33 +85,35 @@ impl PrometheusMiddleware {
                 (None, None)
             };
 
-        let (metrics_push_handle, metrics_push_kill_switch) = if let Some(addr) = pushgateway_addr {
-            let (metrics_push_kill_switch, metrics_push_kill_switch_rx) =
-                tokio::sync::oneshot::channel();
+        let (metrics_push_handle, metrics_push_kill_switch) =
+            if let Some(push_config) = config.pushgateway_config {
+                let (metrics_push_kill_switch, metrics_push_kill_switch_rx) =
+                    tokio::sync::oneshot::channel();
 
-            // need a separate thread for pushing metrics because the reqwest sync client
-            // is used in the prometheus library. reqwest sync creates a tokio runtime internally
-            // so the tokio runtime created for the metrics server can't be reused.
-            let thread_handle = std::thread::spawn(move || {
-                push_metrics(addr, metrics_push_kill_switch_rx);
-            });
+                // need a separate thread for pushing metrics because the reqwest sync client
+                // is used in the prometheus library. reqwest sync creates a tokio runtime internally
+                // so the tokio runtime created for the metrics server can't be reused.
```

Comment:
This statement sums up so much of what I don't like about the current state of async runtimes in Rust. Those kinds of internal details (like reqwest creating a tokio runtime as a block point) should never force consumers to make these kinds of decisions/refactors.